_posts/2020-08-04-newpipe-turning-5.md

```diff
@@ -0,0 +1,31 @@
+I didn't know how to port Python to Android. Therefore, I simply tried to figure out how Youtube-dl does its magic and rewrote the core logic in Java. That code would extract the video stream URL from a simple Youtube URL, and that's it. I wrapped an app around this code. One could use the android share function to enter a link into the app. It would then extract the stream URL and open the video in VLC player. Later I went on and wrote a simple scraper that would also search videos on the website and display the result. When you tap on the result the app would show you some details about the video, like the description and how much up/down votes it got. From here you could simply launch into VLC and watch the video.
+
+That was it. With this simple and slick app I didn't have to use the Youtube mobile website anymore and my problem was solved. I was done. I wanted to release my code on F-Droid to also let others have the opportunity to use it. So I named the app GnuTube and prepared everything for a release. I also added some features like the ability to download a video. Furthermore I set myself a goal of about 10 features/functions I wanted to properly implement before the initial release 1.0, however, as I showed the app to my FSFE friends they said they wouldn't want to wait and I should release it as soon as possible. Somehow, I thought,  'Well, v0.3 would pretty much represent my progress.'
+They also told me "GnuTube" could be a problematic name and is a weird idea as the app wasen't and would never be part of the GNU project. Florian Snow, the founding father of that FSFE fellowship group then suggested me some names from which I picked one more or less at random. Guess what, that name was "NewPipe". He would later tell me that I couldn't have picked a worse name, but by then I already liked it. I found it good because it suggested something "New", and well it was not a "Tube" like in Youtube but a "Pipe" which is funny and prevents name collision. So here we go. on September 4th, 2015 NewPipe came to light, and it was liked pretty much from the beginning on. The way I "measured" popularity was, and still is by today, the stars the project got on GitHub. The stars didn't skyrocket, not like I expected that, but they kept growing and growing ... and growing. They just didn't stop to get more. So I thought maybe I really dug up something here and should keep working on it. I did for the next four years.
```

Comment:
```suggestion
They also told me "GnuTube" could be a problematic name and is a weird idea as the app wasn't and would never be part of the GNU project. Florian Snow, the founding father of that FSFE fellowship group then suggested me some names from which I picked one more or less at random. Guess what, that name was "NewPipe". He would later tell me that I couldn't have picked a worse name, but by then I already liked it. I found it good because it suggested something "New", and well it was not a "Tube" like in YouTube but a "Pipe" which is funny and prevents name collision. So here we go. on September 4th, 2015 NewPipe came to light, and it was liked pretty much from the beginning on. The way I "measured" popularity was, and still is by today, the stars the project got on GitHub. The stars didn't skyrocket, not like I expected that, but they kept growing and growing ... and growing. They just didn't stop to get more. So I thought maybe I really dug up something here and should keep working on it. I did for the next four years.
```

Reply:
that's hard to see what changed when the whole line has lots of words… I noticed the wasen't -> wasn't, but was there something else?

Reply:
I also changed Youtube -> YouTube